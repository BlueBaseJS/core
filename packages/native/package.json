{
  "name": "bluerain-boilerplate-native",
  "version": "0.3.0",
  "description": "Hello BlueRain!",
  "author": "BlueEast Team <code@blueeast.com>",
  "private": true,
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "exp start"
  },
  "dependencies": {
    "@blueeast/bluerain-os": "git://github.com/BlueEastCode/bluerain-os.git#feature/custom_platform",
    "expo": "19.0.1",
    "react": "16.0.0-alpha.12",
    "react-dom": "^16.0.0-alpha.12",
    "react-native": "0.46.4",
    "react-native-windows": "^0.33.0",
    "reactxp": "^0.42.0-rc.25",
    "reactxp-navigation": "1.0.5",
    "reactxp-video": "^0.2.0"
  },
  "devDependencies": {
    "@types/node": "^7.0.12",
    "@types/react": "15.0.39",
    "@types/react-native": "0.46.7",
    "@types/webpack": "^2.2.14",
    "source-map-loader": "^0.1.6"
  }
}
