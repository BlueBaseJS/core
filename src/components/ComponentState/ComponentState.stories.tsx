import { BlueRainType } from '../../index';
import { bluerainStory } from '../../../storybook/bluerain';
import React from 'react';
import storiesOf from '../../../storybook/storiesOf';

storiesOf('ComponentState', module)
	.add('Only title', bluerainStory((props: { bluerain: BlueRainType }) => {

		const BR = props.bluerain;
		const ComponentState = BR.Components.get('ComponentState');

		return <ComponentState title="A simple state" />;
	}))
	.add('title with titleStyle prop', bluerainStory((props: { bluerain: BlueRainType }) => {

		const BR = props.bluerain;
		const ComponentState = BR.Components.get('ComponentState');
		const Style = { color: 'green', fontSize: 20 };

		return <ComponentState title="A simple state" titleStyle={BR.Utils.createStyleSheet(Style)} />;
	}))
	.add('Only description', bluerainStory((props: { bluerain: BlueRainType }) => {

		const BR = props.bluerain;
		const ComponentState = BR.Components.get('ComponentState');

		return <ComponentState description="A simple description" />;
	}))
	.add('description with Styles ', bluerainStory((props: { bluerain: BlueRainType }) => {

		const BR = props.bluerain;
		const ComponentState = BR.Components.get('ComponentState');
		const Style={ color:'green',fontSize:20 };

		return <ComponentState description="A simple description"  descriptionStyle={Style} />;
	}))
	.add('Only buttonTitle', bluerainStory((props: { bluerain: BlueRainType }) => {

		const BR = props.bluerain;
		const ComponentState = BR.Components.get('ComponentState');

		return <ComponentState buttonTitle="Bluerain"  />;
	}))
	.add('Button with buttonOnPress Prop', bluerainStory((props: { bluerain: BlueRainType }) => {
		const Handler=() => console.log('clicked');

		const BR = props.bluerain;
		const ComponentState = BR.Components.get('ComponentState');

		return <ComponentState   buttonTitle="Bluerain" buttonOnPress={Handler}  />;
	}))
	.add('Button with buttonStyle prop', bluerainStory((props: { bluerain: BlueRainType }) => {
		const BR = props.bluerain;
		const ComponentState = BR.Components.get('ComponentState');
		const Style={ color:'white',backgroundColor:'red',width:100,height:50,borderRadius:8, textAlign:'center' };

		return <ComponentState  buttonTitle="Bluerain"  buttonStyle={BR.Utils.createStyleSheet(Style)}  />;

	}))
	.add('Button Component', bluerainStory((props: { bluerain: BlueRainType }) => {

		const Button=() =>  <button>ButtonComponent</button>;
		const BR = props.bluerain;
		const ComponentState = BR.Components.get('ComponentState');
		// const Style={ color:'white',backgroundColor:'red',width:100,height:50,borderRadius:8, textAlign:'center' };

		return <ComponentState  button={Button}  />;
	}))
	.add('imageComponent prop', bluerainStory((props: { bluerain: BlueRainType }) => {

		const BR = props.bluerain;
		const Image = BR.Components.get('Image');
		const Style = { width: 200, height: 200 };
		// tslint:disable-next-line:max-line-length
		const ImageComponent = () => <Image source="https://lh3.googleusercontent.com/0-BzaWtxoAnsBjQ_wzUcKxyF07XE7v2Kkg1ogPVUdzmQpvaz118uHQEGU6BdtzJuzfo=h1264" imageStyle={BR.Utils.createStyleSheet(Style)} />;
		const ComponentState = BR.Components.get('ComponentState');
		return <ComponentState image={ImageComponent} />;
	}))

	.add('imageSource prop', bluerainStory((props: { bluerain: BlueRainType }) => {

		const BR = props.bluerain;
		// const Image = BR.Components.get('Image');
		const Style={ width:200,height:200 };
		const ComponentState = BR.Components.get('ComponentState');
		// tslint:disable-next-line:max-line-length
		return <ComponentState  imageSource="https://lh3.googleusercontent.com/0-BzaWtxoAnsBjQ_wzUcKxyF07XE7v2Kkg1ogPVUdzmQpvaz118uHQEGU6BdtzJuzfo=h1264"  imageStyle={BR.Utils.createStyleSheet(Style)} />;
	}));

// storiesOf('ComponentState', module).add('Simple', () => <div>hello </div>);
