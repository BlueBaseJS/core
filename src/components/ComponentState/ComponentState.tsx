import React from 'react';
import { ViewProperties, ImageStyle, ViewStyle, TextStyle } from '@blueeast/bluerain-ui-interfaces';
import { withBlueRain, BlueRainType } from '../../index';

export interface ComponentStateProps extends ViewProperties {
	/**
	 * Image Component, if provided, imageSource will be ignored
	 */
	image?: React.ComponentType;

	/**
	 * Image styles
	 */
	imageStyle?: ImageStyle;

	/**
	 * Image source
	 */
	imageSource?: string;

	/**
	 * Title text
	 */
	title: string;

	/**
	 * Title style
	 */
	titleStyle?: TextStyle;

	/**
	 * Description Text
	 */
	description: string;

	/**
	 * Description style
	 */
	descriptionStyle?: TextStyle;

	/**
	 * Button Component, if provided, other button props will be ignored
	 */
	button?: React.ComponentType;

	/**
	 * Button title
	 */
	buttonTitle?: string;

	/**
	 * Button styles
	 */
	buttonStyle?: ViewStyle;

	/**
	 * Button onPress handler
	 */
	buttonOnPress?: Function;

	style:ViewStyle
}

const ComponentState = (props: ComponentStateProps & { bluerain: BlueRainType }) => {
	const {
		image: ImageComponent,
		imageSource,
		title,
		description,
		button: ButtonComponent,
		buttonTitle,
		buttonOnPress,
		bluerain: BR
	} = props;
	const View = BR.Components.get('View');
	const Text = BR.Components.get('Text');
	const Image = BR.Components.get('Image');
	const Button = BR.Components.get('Button');

	const style = {
		alignItems: 'center',
		padding: 16,
	};

	const stylesheet = [
		BR.Utils.createStyleSheet(style),
		BR.Utils.createStyleSheet(props.style || {})
	];

	const titleStyle = BR.Utils.createStyleSheet(props.titleStyle || {});
	const descriptionStyle = BR.Utils.createStyleSheet(props.descriptionStyle || {});

	// Image
	const imageStyle = [
		BR.Utils.createStyleSheet({ width: 100 }),
		BR.Utils.createStyleSheet(props.imageStyle || {}),
	];

	let ImageC = ImageComponent || null;
	console.log(ImageC,imageSource);

	if (!ImageC && imageSource) {
		console.log('aaakakakakakakkkkakakakakakkkakakak');
		ImageC = () => (<Image style={imageStyle} source={imageSource} />);
	}

	// Button
	const buttonStyle = [
		BR.Utils.createStyleSheet({ paddingTop: 16 }),
		BR.Utils.createStyleSheet(props.buttonStyle || {}),
	];

	let ButtonC = ButtonComponent || null;
	if (!ButtonC && buttonTitle) {
		ButtonC = () => (<Button style={buttonStyle} onPress={buttonOnPress}><Text>{buttonTitle}</Text></Button>);
	}

	return (
  <View style={stylesheet}>
    {(ImageC) ? <ImageC /> : null}
    {(title) ? <Text style={titleStyle} >{title}</Text> : null}
    {(description) ? <Text style={descriptionStyle} >{description}</Text> : null}
    {(ButtonC) ? <ButtonC /> : null}
  </View>
	);
};

export default withBlueRain(ComponentState);
