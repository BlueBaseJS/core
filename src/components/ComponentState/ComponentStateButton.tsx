import { BlueRainType,withBlueRain } from '../../index';
import { ImageStyle, TextStyle,ViewProperties,ViewStyle } from '@blueeast/bluerain-ui-interfaces';
import React from 'react';

export interface ComponentStateButtonProps extends ViewProperties {
	/**
	 * Button Component, if provided, other button props will be ignored
	 */
	button?: React.ComponentType;

	/**
	 * Button title
	 */
	buttonTitle?: string;

	/**
	 * Button styles
	 */
	buttonStyle?: ViewStyle;

	/**
	 * Button onPress handler
	 */
	buttonOnPress?: Function;

	bluerain: BlueRainType;
}

const ComponentStateButton = (props: ComponentStateButtonProps) => {

	const {
		button: ButtonComponent,
		buttonTitle,
		buttonOnPress,
		buttonStyle,
		bluerain: BR
	} = props;

	if (ButtonComponent) {

		return (<ButtonComponent />);

	} else if (buttonTitle) {
     	console.log(BR);
		const Button = BR.Components.get('Button');
		const Text = BR.Components.get('Text');

		const stylesheet = {
			marginTop: 10,
			...buttonStyle
		};

		return (
			<Button onPress={buttonOnPress} style={BR.Utils.createStyleSheet(stylesheet)}>
				<Text>{buttonTitle}</Text>
			</Button>
		);

	} else {
		return null;
	}
};

export default withBlueRain(ComponentStateButton);
