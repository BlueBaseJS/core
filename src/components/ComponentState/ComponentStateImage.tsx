import { BlueRainType ,withBlueRain } from '../../index';
import { ImageStyle,TextStyle,ViewProperties , ViewStyle } from '@blueeast/bluerain-ui-interfaces';
import React from 'react';

export interface ComponentStateImageProps extends ViewProperties {
	/**
	 * Image Component, if provided, imageSource will be ignored
	 */
	image?: React.ComponentType;

	/**
	 * Image styles
	 */
	imageStyle?: ImageStyle;

	/**
	 * Image source
	 */
	imageSource?: string;

	style?: ViewStyle;

	bluerain: BlueRainType;
}

const ComponentStateImage = (props: ComponentStateImageProps) => {
	const {
		image: ImageComponent,
		imageSource,
		imageStyle,
		bluerain: BR
	} = props;

	const View = BR.Components.get('View');
	const Image = BR.Components.get('Image');

	// Image
	let ImageC;

	if (ImageComponent || imageSource) {
		ImageC = ImageComponent || Image;

	} else {
		return null;
	}

	// Styles
	const stylesheet = { marginBottom: 10, maxWidth: 200 };

	return (
  <View style={BR.Utils.createStyleSheet(stylesheet)}>
		<ImageC
			style={imageStyle ? BR.Utils.createStyleSheet(imageStyle) : {}}
			source={imageSource}
		/>
  </View>
	);
};

export default withBlueRain(ComponentStateImage);
