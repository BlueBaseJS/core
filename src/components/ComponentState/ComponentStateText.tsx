import { BlueRain } from '../../index';
import { TextStyle } from '@blueeast/bluerain-ui-interfaces';
import React from 'react';

export interface ComponentStateTextProps {
	/**
	 * Text
	 */
	text: string;

	/**
	 * Style Object
	 */
	style?: TextStyle;

	bluerain: BlueRain;
}

const ComponentStateText = (props: ComponentStateTextProps) => {

	const {
		text,
		style,
		bluerain: BR
	} = props;

	if (text) {

		const Text = BR.Components.get('Text');

		return (
			<Text style={style ? BR.Utils.createStyleSheet(style, 'Text') : {}} >
				{text}
			</Text>
		);

	} else {
		return null;
	}
};

export default ComponentStateText;
