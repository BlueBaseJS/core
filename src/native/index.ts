export * from './ActivityIndicator';
export * from './Button';
export * from './Image';
export * from './Text';
export * from './View';
