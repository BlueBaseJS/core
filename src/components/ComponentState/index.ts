import ComponentState from './ComponentState';
export * from './ComponentState';
export default ComponentState;
