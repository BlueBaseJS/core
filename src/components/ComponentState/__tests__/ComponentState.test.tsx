import * as Native from '../../../native';

import { BlueBaseApp } from '../../../';
import { ComponentState } from '../ComponentState';
import React from 'react';
import { mount } from 'enzyme';
import { waitForElement } from 'enzyme-async-helpers';

describe('ComponentState', () => {
	test(`should render ComponentState with all elements`, async () => {
		const wrapper = mount(
			<BlueBaseApp>
				<ComponentState
					title="Looks like your'e new here!"
					description="Start by creating your first entry."
					imageSource={{ uri: 'https://picsum.photos/200' }}
					styles={{ image: { width: 100, height: 100 } }}
					actionTitle="Tap to Create"
				/>
			</BlueBaseApp>
		);

		// Wait for render
		await waitForElement(wrapper as any, ComponentState);

		// Check Component
		expect(
			wrapper
				.find('ComponentState Image')
				.last()
				.prop('source')
		).toMatchObject({
			uri: 'https://picsum.photos/200',
		});

		const styles = wrapper
			.find('ComponentState Image')
			.last()
			.prop('style');
		if (styles) {
			expect(styles.height).toBe(100);
			expect(styles.width).toBe(100);
		}
		expect(
			wrapper
				.find('ComponentState H6 Text')
				.last()
				.text()
		).toBe("Looks like your'e new here!");
		expect(
			wrapper
				.find('ComponentState Body2 Text')
				.last()
				.text()
		).toBe('Start by creating your first entry.');
		expect(
			wrapper
				.find('ComponentState Button Text')
				.last()
				.text()
		).toBe('Tap to Create');
	});

	test(`should render ComponentState with custom image`, async () => {
		const wrapper = mount(
			<BlueBaseApp>
				<ComponentState
					image={
						<Native.Image
							testID="Custom"
							style={{ width: 50, height: 50 }}
							source={{ uri: 'https://picsum.photos/200' }}
						/>
					}
					styles={{ image: { width: 100, height: 100 } }}
				/>
			</BlueBaseApp>
		);

		// Wait for render
		await waitForElement(wrapper as any, ComponentState);

		// Check Component
		expect(
			wrapper
				.find('ComponentState Image')
				.last()
				.prop('source')
		).toMatchObject({
			uri: 'https://picsum.photos/200',
		});
		expect(
			wrapper
				.find('ComponentState Image')
				.last()
				.prop('style')
		).toMatchObject({
			height: 50,
			width: 50,
		});
		expect(
			wrapper
				.find('ComponentState Image')
				.last()
				.prop('testID')
		).toBe('Custom');
		expect(wrapper.exists('ComponentState H6')).toBe(false);
		expect(wrapper.exists('ComponentState Body2')).toBe(false);
		expect(wrapper.exists('ComponentState Button')).toBe(false);
	});
});
