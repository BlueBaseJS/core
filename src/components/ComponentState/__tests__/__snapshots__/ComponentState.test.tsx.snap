// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentState should render ComponentState with all elements 1`] = `
<BlueBaseApp>
  <BlueBaseRoot>
    <ThemeProvider>
      <SystemApp>
        <ThemedComponent>
          <SystemApp>
            <ComponentState
              actionTitle="Tap to Create"
              description="Start by creating your first entry."
              imageSource="https://picsum.photos/200"
              styles={
                Object {
                  "image": Object {
                    "height": 100,
                    "width": 100,
                  },
                }
              }
              title="Looks like your'e new here!"
            >
              <View>
                <ThemedComponent>
                  <View>
                    <View>
                      <View>
                        <ThemedComponent>
                          <View>
                            <View>
                              <Image
                                source={
                                  Object {
                                    "uri": "https://picsum.photos/200",
                                  }
                                }
                                style={
                                  Object {
                                    "height": 100,
                                    "width": 100,
                                  }
                                }
                              >
                                <ThemedComponent
                                  source={
                                    Object {
                                      "uri": "https://picsum.photos/200",
                                    }
                                  }
                                  style={
                                    Object {
                                      "height": 100,
                                      "width": 100,
                                    }
                                  }
                                >
                                  <Component
                                    source={
                                      Object {
                                        "uri": "https://picsum.photos/200",
                                      }
                                    }
                                    style={
                                      Object {
                                        "height": 100,
                                        "width": 100,
                                      }
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "uri": "https://picsum.photos/200",
                                        }
                                      }
                                      style={
                                        Object {
                                          "height": 100,
                                          "width": 100,
                                        }
                                      }
                                    />
                                  </Component>
                                </ThemedComponent>
                              </Image>
                            </View>
                          </View>
                        </ThemedComponent>
                      </View>
                      <H6>
                        <ThemedComponent>
                          <Component
                            styles={
                              Object {
                                "root": Object {
                                  "color": "rgba(0, 0, 0, 0.87)",
                                  "fontFamily": "Helvetica",
                                  "fontSize": 20,
                                  "fontWeight": "500",
                                },
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "rgba(0, 0, 0, 0.87)",
                                    "fontFamily": "Helvetica",
                                    "fontSize": 20,
                                    "fontWeight": "500",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "rgba(0, 0, 0, 0.87)",
                                      "fontFamily": "Helvetica",
                                      "fontSize": 20,
                                      "fontWeight": "500",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Looks like your'e new here!
                              </Text>
                            </Text>
                          </Component>
                        </ThemedComponent>
                      </H6>
                      <Body2>
                        <ThemedComponent>
                          <Component
                            styles={
                              Object {
                                "root": Object {
                                  "color": "rgba(0, 0, 0, 0.87)",
                                  "fontFamily": "Helvetica",
                                  "fontSize": 14,
                                  "fontWeight": "400",
                                },
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "rgba(0, 0, 0, 0.87)",
                                    "fontFamily": "Helvetica",
                                    "fontSize": 14,
                                    "fontWeight": "400",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "rgba(0, 0, 0, 0.87)",
                                      "fontFamily": "Helvetica",
                                      "fontSize": 14,
                                      "fontWeight": "400",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Start by creating your first entry.
                              </Text>
                            </Text>
                          </Component>
                        </ThemedComponent>
                      </Body2>
                      <View>
                        <ThemedComponent>
                          <View>
                            <View>
                              <Button
                                color="primary"
                              >
                                <ThemedComponent
                                  color="primary"
                                >
                                  <Button
                                    color="primary"
                                    styles={
                                      Object {
                                        "default": Object {
                                          "backgroundColor": "#fafafa",
                                        },
                                        "defaultText": Object {},
                                        "fullWidth": Object {
                                          "alignSelf": "stretch",
                                        },
                                        "link": Object {},
                                        "linkText": Object {},
                                        "primary": Object {
                                          "backgroundColor": "#3f51b5",
                                        },
                                        "primaryText": Object {
                                          "color": "#fff",
                                        },
                                        "root": Object {
                                          "alignSelf": "flex-start",
                                          "borderRadius": 4,
                                          "paddingBottom": 8,
                                          "paddingLeft": 16,
                                          "paddingRight": 16,
                                          "paddingTop": 8,
                                        },
                                        "secondary": Object {
                                          "backgroundColor": "#f50057",
                                        },
                                        "secondaryText": Object {
                                          "color": "#fff",
                                        },
                                        "text": Object {
                                          "color": "rgba(0, 0, 0, 0.87)",
                                          "fontFamily": "Helvetica",
                                          "fontSize": 14,
                                          "fontWeight": "500",
                                          "textAlign": "center",
                                          "textTransform": "uppercase",
                                        },
                                      }
                                    }
                                  >
                                    <TouchableOpacity
                                      accessibilityRole="button"
                                      activeOpacity={0.2}
                                    >
                                      <AnimatedComponent
                                        accessibilityRole="button"
                                        accessible={true}
                                        isTVSelectable={true}
                                        onResponderGrant={[Function]}
                                        onResponderMove={[Function]}
                                        onResponderRelease={[Function]}
                                        onResponderTerminate={[Function]}
                                        onResponderTerminationRequest={[Function]}
                                        onStartShouldSetResponder={[Function]}
                                        style={
                                          Array [
                                            undefined,
                                            Object {
                                              "opacity": 1,
                                            },
                                          ]
                                        }
                                      >
                                        <View
                                          accessibilityRole="button"
                                          accessible={true}
                                          isTVSelectable={true}
                                          onResponderGrant={[Function]}
                                          onResponderMove={[Function]}
                                          onResponderRelease={[Function]}
                                          onResponderTerminate={[Function]}
                                          onResponderTerminationRequest={[Function]}
                                          onStartShouldSetResponder={[Function]}
                                          style={
                                            Object {
                                              "opacity": 1,
                                            }
                                          }
                                        >
                                          <View
                                            accessibilityRole="button"
                                            accessible={true}
                                            isTVSelectable={true}
                                            onResponderGrant={[Function]}
                                            onResponderMove={[Function]}
                                            onResponderRelease={[Function]}
                                            onResponderTerminate={[Function]}
                                            onResponderTerminationRequest={[Function]}
                                            onStartShouldSetResponder={[Function]}
                                            style={
                                              Object {
                                                "opacity": 1,
                                              }
                                            }
                                          >
                                            <View
                                              style={
                                                Array [
                                                  Object {
                                                    "alignSelf": "flex-start",
                                                    "borderRadius": 4,
                                                    "paddingBottom": 8,
                                                    "paddingLeft": 16,
                                                    "paddingRight": 16,
                                                    "paddingTop": 8,
                                                  },
                                                  Object {
                                                    "backgroundColor": "#3f51b5",
                                                  },
                                                ]
                                              }
                                            >
                                              <View
                                                style={
                                                  Array [
                                                    Object {
                                                      "alignSelf": "flex-start",
                                                      "borderRadius": 4,
                                                      "paddingBottom": 8,
                                                      "paddingLeft": 16,
                                                      "paddingRight": 16,
                                                      "paddingTop": 8,
                                                    },
                                                    Object {
                                                      "backgroundColor": "#3f51b5",
                                                    },
                                                  ]
                                                }
                                              >
                                                <Text
                                                  style={
                                                    Array [
                                                      Object {
                                                        "color": "rgba(0, 0, 0, 0.87)",
                                                        "fontFamily": "Helvetica",
                                                        "fontSize": 14,
                                                        "fontWeight": "500",
                                                        "textAlign": "center",
                                                        "textTransform": "uppercase",
                                                      },
                                                      Object {
                                                        "color": "#fff",
                                                      },
                                                    ]
                                                  }
                                                >
                                                  <Text
                                                    style={
                                                      Array [
                                                        Object {
                                                          "color": "rgba(0, 0, 0, 0.87)",
                                                          "fontFamily": "Helvetica",
                                                          "fontSize": 14,
                                                          "fontWeight": "500",
                                                          "textAlign": "center",
                                                          "textTransform": "uppercase",
                                                        },
                                                        Object {
                                                          "color": "#fff",
                                                        },
                                                      ]
                                                    }
                                                  >
                                                    Tap to Create
                                                  </Text>
                                                </Text>
                                              </View>
                                            </View>
                                          </View>
                                        </View>
                                      </AnimatedComponent>
                                    </TouchableOpacity>
                                  </Button>
                                </ThemedComponent>
                              </Button>
                            </View>
                          </View>
                        </ThemedComponent>
                      </View>
                    </View>
                  </View>
                </ThemedComponent>
              </View>
            </ComponentState>
          </SystemApp>
        </ThemedComponent>
      </SystemApp>
    </ThemeProvider>
  </BlueBaseRoot>
</BlueBaseApp>
`;

exports[`ComponentState should render ComponentState with custom image 1`] = `
<BlueBaseApp>
  <BlueBaseRoot>
    <ThemeProvider>
      <SystemApp>
        <ThemedComponent>
          <SystemApp>
            <ComponentState
              image={
                <Component
                  source={
                    Object {
                      "uri": "https://picsum.photos/200",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                  testID="Custom"
                />
              }
              styles={
                Object {
                  "image": Object {
                    "height": 100,
                    "width": 100,
                  },
                }
              }
            >
              <View>
                <ThemedComponent>
                  <View>
                    <View>
                      <View>
                        <ThemedComponent>
                          <View>
                            <View>
                              <Component
                                source={
                                  Object {
                                    "uri": "https://picsum.photos/200",
                                  }
                                }
                                style={
                                  Object {
                                    "height": 50,
                                    "width": 50,
                                  }
                                }
                                testID="Custom"
                              >
                                <Image
                                  source={
                                    Object {
                                      "uri": "https://picsum.photos/200",
                                    }
                                  }
                                  style={
                                    Object {
                                      "height": 50,
                                      "width": 50,
                                    }
                                  }
                                  testID="Custom"
                                />
                              </Component>
                            </View>
                          </View>
                        </ThemedComponent>
                      </View>
                    </View>
                  </View>
                </ThemedComponent>
              </View>
            </ComponentState>
          </SystemApp>
        </ThemedComponent>
      </SystemApp>
    </ThemeProvider>
  </BlueBaseRoot>
</BlueBaseApp>
`;
